File: 4.8-bc/ts/stage-1-flat/src/repository/order.ts
import type { DbPool } from "../db.js";
import type { Order } from "../model/order.js";

export class OrderRepository {
  constructor(private readonly db: DbPool) {}

  async create(o: Order): Promise<void> {
    await this.db.query(
      `INSERT INTO orders (id, user_id, total_cents, status, created_at)
       VALUES ($1, $2, $3, $4, $5)`,
      [o.id, o.userId, o.totalCents, o.status, o.createdAt],
    );
  }
}

File: 4.8-bc/py/stage-1-flat/app/repository/order.py
from __future__ import annotations

from sqlalchemy import text
from sqlalchemy.ext.asyncio import AsyncSession

from app.model.order import Order


class OrderRepo:
    def __init__(self, session: AsyncSession) -> None:
        self._session = session

    async def create(self, o: Order) -> None:
        await self._session.execute(
            text(
                "INSERT INTO orders (id, user_id, total_cents, status, created_at) "
                "VALUES (:id, :user_id, :total_cents, :status, :created_at)"
            ),
            {
                "id": o.id,
                "user_id": o.user_id,
                "total_cents": o.total_cents,
                "status": o.status,
                "created_at": o.created_at,
            },
        )
        await self._session.commit()
